src/ngraph/op/divide.cpp

```diff
@@ -78,6 +79,60 @@ shared_ptr<Node> ngraph::operator/(const Output<Node>& arg0, const Output<Node>&
     return make_shared<op::v0::Divide>(arg0, arg1);
 }
 
+namespace
+{
+    template <element::Type_t ET>
+    bool try_evaluate_divide(const EvaluatorTensorPtr& arg0,
+                             const EvaluatorTensorPtr& arg1,
+                             const EvaluatorTensorPtr& out,
+                             const op::AutoBroadcastSpec& broadcast_spec,
+                             bool pythondiv)
+    {
+        return (ET == arg0->get_element_type()) && (runtime::reference::divide(arg0->get_ptr<ET>(),
+                                                                               arg1->get_ptr<ET>(),
+                                                                               out->get_ptr<ET>(),
+                                                                               arg0->get_shape(),
+                                                                               arg1->get_shape(),
+                                                                               broadcast_spec,
+                                                                               pythondiv),
+                                                    true);
+    }
+
+    bool evaluate_divide(const EvaluatorTensorPtr& arg0,
+                         const EvaluatorTensorPtr& arg1,
+                         const EvaluatorTensorPtr& out,
+                         const op::AutoBroadcastSpec& broadcast_spec,
+                         bool pythondiv)
+    {
+        return try_evaluate_divide<element::Type_t::i8>(
+                   arg0, arg1, out, broadcast_spec, pythondiv) ||
+               try_evaluate_divide<element::Type_t::i16>(
+                   arg0, arg1, out, broadcast_spec, pythondiv) ||
+               try_evaluate_divide<element::Type_t::i32>(
+                   arg0, arg1, out, broadcast_spec, pythondiv) ||
+               try_evaluate_divide<element::Type_t::i64>(
+                   arg0, arg1, out, broadcast_spec, pythondiv) ||
+               try_evaluate_divide<element::Type_t::u8>(
+                   arg0, arg1, out, broadcast_spec, pythondiv) ||
+               try_evaluate_divide<element::Type_t::u16>(
+                   arg0, arg1, out, broadcast_spec, pythondiv) ||
+               try_evaluate_divide<element::Type_t::u32>(
+                   arg0, arg1, out, broadcast_spec, pythondiv) ||
+               try_evaluate_divide<element::Type_t::u64>(
+                   arg0, arg1, out, broadcast_spec, pythondiv) ||
+               try_evaluate_divide<element::Type_t::f32>(
+                   arg0, arg1, out, broadcast_spec, pythondiv) ||
+               try_evaluate_divide<element::Type_t::f64>(
+                   arg0, arg1, out, broadcast_spec, pythondiv);
```

Comment:
Use a switch statement here because
1) It is easier to read and understand
2) It allows for compile-time checking that all types are supported
3) Allows a more informative error message if something goes wrong, currently any failure returns false. It might be nice for more information why something failed

This statement applies to all ops below